sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuExec.scala

```diff
@@ -103,25 +103,14 @@ object GpuMetric extends Logging {
   val DESCRIPTION_READ_FS_TIME = "time to read fs data"
   val DESCRIPTION_WRITE_BUFFER_TIME = "time to write data to buffer"
 
-  def unwrap(input: GpuMetric): SQLMetric = input match {
-    case w :WrappedGpuMetric => w.sqlMetric
-    case i => throw new IllegalArgumentException(s"found unsupported GpuMetric ${i.getClass}")
+  def unwrap(input: Map[String, GpuMetric]): Map[String, SQLMetric] = input.collect {
```

Comment:
Not sure whether this is ok, since the new version will ignore the unsupported GpuMetric classes, but the old one can blow it up.

Reply:
> but the old one can blow it up.

This is not the truth, I guess. Although we have the branch
```scala
case i => throw new IllegalArgumentException(s"found unsupported GpuMetric ${i.getClass}")
```
This branch will be never touched.

Because in the original code, we first filter out `NoopMetrics` by doing this
```scala
  def unwrap(input: Map[String, GpuMetric]): Map[String, SQLMetric] = input.filter {
    // remove the metrics that are not registered
    case (_, NoopMetric) => false
    case _ => true
    // sadly mapValues produces a non-serializable result, so we have to hack it a bit to force
    // it to be materialized
  }.mapValues(unwrap).toArray.toMap
```
So we can make sure that when calling `mapValues(unwrap)`, all values have the type `WrappedGpuMetric`. This is the reason that we don't need to the pattern matching again in `def unwrap(input: GpuMetric): SQLMetric`
```

Reply:
> This unwrap function is public and can be called directly.

Reply:
> This unwrap function is public and can be called directly.

Thank you for the clarification. I could bring this `unwrap` back, although I don't find we use this function in any other place. But we should not use this `unwrap` function in `def unwrap(input: Map[String, GpuMetric]): Map[String, SQLMetric]`.

Reply:
Hi @jlowe, I bring that `unwrap` function back, could you please help to review?